test/integration/test_messages_api.py

```diff
@@ -0,0 +1,187 @@
+            to=['+invalid'],
+            _from='+invalid',
+            text='',
+        )
+        
+        with self.assertRaises(ApiException) as context:
+            self.api_instance.create_message(account_id, message_request)
+            
+        self.assertEqual(context.exception.status, 400)
+
+        e = CreateMessageRequestError(field_errors=(json.loads(context.exception.body))['fieldErrors'], **json.loads(context.exception.body))
+        self.assertIsInstance(e, CreateMessageRequestError)
+        self.assertEqual(e.type, 'request-validation')
+        self.assertIsInstance(e.description, str)
+        self.assertIsInstance(e.field_errors, list)
+
+        error = e.field_errors[0]
+        self.assertIsInstance(error, FieldError)
+        self.assertIsInstance(error.fieldName, str)
+        self.assertIsInstance(error.description, str)
+    
+
+    def test_create_message_unauthorized(self):
+        api_client = bandwidth.ApiClient()
+        api_instance = messages_api.MessagesApi(api_client)
+        account_id = os.environ.get('BW_ACCOUNT_ID')
+        message_request = MessageRequest(
+            application_id=os.environ.get('BW_MESSAGING_APPLICATION_ID'),
+            to=['+invalid'],
+            _from='+invalid',
+            text='',
+        )
+        
+        with self.assertRaises(UnauthorizedException) as context:
+            api_instance.create_message(account_id, message_request)
+             
+        self.assertEqual(context.exception.status, 401)
+
+        e = MessagingRequestError(**json.loads(context.exception.body))
+        self.assertIsInstance(e, MessagingRequestError)
+        self.assertEqual(e.type, 'unauthorized')
+        self.assertEqual(e.description, 'Authentication Failed')
+
+    # The SDK catches incorrect content-type before making the request and attempts to create an ApiException,
+    # but the creation of the exception fails since there is no response body. This should probably create some
+    # kind of Client Exception instead, since this isn't an actual API Exception.
+
+    def test_list_messages(self):
+        account_id = os.environ.get('BW_ACCOUNT_ID')
+        message_direction = ListMessageDirectionEnum("OUTBOUND")
+        
+        try:
+            api_response = self.api_instance.list_messages(account_id, message_direction=message_direction)
+        except bandwidth.ApiException as e:
+            print("Exception when calling MessagesApi->list_messages: %s\n" % e)
+
+        self.assertIsInstance(api_response, MessagesList)
+        self.assertGreater(api_response.total_count, 0)
+        self.assertTrue(api_response.messages[0], ListMessageItem)
+        
+        message = api_response.messages[0]
+        self.assertEqual(message.account_id, os.environ.get('BW_ACCOUNT_ID'))
+        self.assertTrue(message.carrier_name)
+        self.assertRegex(message.destination_tn, '^\\+[1-9]\\d{1,14}$')
```

Comment:
Why a regex string when we have the exact value it should be?

Reply:
we don't have the exact value guaranteed. It's listing all messages from our account and there's a good chance it'll be the message sent in these tests, but it's not guaranteed

Reply:
ah true - feel free to resolve 👍